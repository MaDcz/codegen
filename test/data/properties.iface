interface Type1
{
  bool   bool_field;
  bool[] bool_list_field;

  int   int_field;
  int[] int_list_field;

  uint   uint_field;
  uint[] uint_list_field;
}

interface Type2
{
  Type1   type1_field;
  Type1[] type1_list_field;
}
