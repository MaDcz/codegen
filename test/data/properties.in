{"type": "codemodel.classdiagram.Package", "nodes": [{"type": "codemodel.classdiagram.Class", "attrs": {"name": "Type1"}, "nodes": [{"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["bool"], "name": "bool_field", "is_repeated": false}}, {"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["bool"], "name": "bool_list_field", "is_repeated": true}}, {"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["int"], "name": "int_field", "is_repeated": false}}, {"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["int"], "name": "int_list_field", "is_repeated": true}}, {"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["uint"], "name": "uint_field", "is_repeated": false}}, {"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["uint"], "name": "uint_list_field", "is_repeated": true}}]}, {"type": "codemodel.classdiagram.Class", "attrs": {"name": "Type2"}, "nodes": [{"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["Type1"], "name": "type1_field", "is_repeated": false}}, {"type": "codemodel.classdiagram.Attribute", "attrs": {"type": ["Type1"], "name": "type1_list_field", "is_repeated": true}}]}]}
