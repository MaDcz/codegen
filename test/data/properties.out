#include <mad/codegen/compositeproperty.hpp>
#include <mad/codegen/compositeslistproperty.hpp>
#include <mad/codegen/tree/compositenode.hpp>
#include <mad/codegen/valueproperty.hpp>
#include <mad/codegen/valueslistproperty.hpp>
class Type1 : public mad::codegen::tree::CompositeNode
{
public:
  Type1();
  virtual ~Type1();
public:
  mad::codegen::ValueProperty<bool> bool_field;
  mad::codegen::ValuesListProperty<bool> bool_list_field;
  mad::codegen::ValueProperty<int> int_field;
  mad::codegen::ValuesListProperty<int> int_list_field;
  mad::codegen::ValueProperty<unsigned int> uint_field;
  mad::codegen::ValuesListProperty<unsigned int> uint_list_field;
};
class Type2 : public mad::codegen::tree::CompositeNode
{
public:
  Type2();
  virtual ~Type2();
public:
  mad::codegen::CompositeProperty<Type1> type1_field;
  mad::codegen::CompositesListProperty<Type1> type1_list_field;
};
Type1::Type1()
  : bool_field(*this, "bool_field"),
    bool_list_field(*this, "bool_list_field"),
    int_field(*this, "int_field"),
    int_list_field(*this, "int_list_field"),
    uint_field(*this, "uint_field"),
    uint_list_field(*this, "uint_list_field"){
}
Type1::~Type1()
{
}
Type2::Type2()
  : type1_field(*this, "type1_field"),
    type1_list_field(*this, "type1_list_field"){
}
Type2::~Type2()
{
}
